core/src/avm2/method.rs

```diff
@@ -301,21 +302,87 @@ impl<'gc> fmt::Debug for NativeMethod<'gc> {
     }
 }
 
+#[derive(Clone, Collect, Debug, Copy, PartialEq, Eq)]
+#[collect(no_drop)]
+pub enum MethodPosition {
+    ClassTrait,
+    InstanceTrait,
+}
+
+#[derive(Clone, Collect, Debug, Copy, PartialEq, Eq)]
+#[collect(no_drop)]
+pub enum MethodKind {
+    Initializer,
+    Setter,
+    Getter,
+    Regular,
+}
+
+#[derive(Clone, Collect, Debug, Copy)]
+#[collect(no_drop)]
+pub struct MethodMetadata<'gc> {
+    name: AvmString<'gc>,
+    position: MethodPosition,
+    kind: MethodKind,
+}
+
+impl<'gc> MethodMetadata<'gc> {
+    pub fn from_trait(my_trait: &Trait<'gc>, position: MethodPosition) -> Self {
+        Self {
+            name: my_trait.name().local_name(),
```

Comment:
Looks like it's the full name (not sure if it's a "qualified name" though)
```
class C {
    ns function f(){ throw new Error(); }
}
```
shows
```
Error
	at C/asdf::f()
	at global$init()
```

Reply:
Looks like it shows the full name when it is a runtime qname, will look into it more later